### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Sbavdik**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,14 @@
...
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeople = century ? people.filter(person =>
+    century === Math.ceil(person.died / 100) && person.sex === 'm')
+    : people.filter(person => person.sex === 'm');
```
Comment (liliia-livandovskaya) [inline src/averageAges.js:20]:
```suggestion
  const filteredPeople = century
    ? people.filter(person =>
         century === Math.ceil(person.died / 100) && person.sex === 'm')
    : people.filter(person => person.sex === 'm');
```

File: src/averageAges.js [hunk 2]
```diff
@@ -55,7 +72,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findMom = onlyWithSon
+    ? people
+      .filter(person => people.some(mother => mother.name === person.mother)
+       && person.sex === 'm')
```
Comment (liliia-livandovskaya) [inline src/averageAges.js:78]:
```suggestion
         && person.sex === 'm')
```

File: src/averageAges.js [hunk 3]
```diff
@@ -37,7 +41,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people
+      .filter(person => person.mother !== null)
```
Comment (SerhiiKirik) [inline src/averageAges.js:46]: A lot of actions
You can just find women with children or women & calc average)

Comment (SerhiiKirik) [reply src/averageAges.js:46]: Try to make better

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Well done 🚀 
Let's improve your code a bit 🧑‍💻

**Review by SerhiiKirik — APPROVED** [review]
Great 🚀

**Pull request closed without merging**